### PR TITLE
#589: Make the constructors of FkHitRefresh compliant with ConstructorOnlyInitializesOrCallOtherConstructors

### PR DESCRIPTION
**Pull request by essobedo**
Fix for https://github.com/yegor256/takes/issues/589

The goal of this PR is to make the class FkHitRefresh compliant with the rule ConstructorOnlyInitializesOrCallOtherConstructors by delegating the construction of the file to touch to a private static method.

### CODE REVIEW

**Comment by davvd** [comment]
@essobedo Thanks for the pull request, let me find a reviewer..

**Comment by davvd** [comment]
@longtimeago review this one, please

File: src/main/java/org/takes/facets/fork/FkHitRefresh.java [hunk 1]
```diff
@@ -172,7 +164,16 @@ private boolean expired() {
      * @throws IOException If fails
      */
     private void touch() throws IOException {
```
Comment (longtimeago) [inline src/main/java/org/takes/facets/fork/FkHitRefresh.java]: @essobedo looks like we don't need this method anymore

**Comment by longtimeago** [comment]
@essobedo see one remark above, thanks!

File: src/main/java/org/takes/facets/fork/FkHitRefresh.java (hunk 1 above)
Comment (essobedo) [inline src/main/java/org/takes/facets/fork/FkHitRefresh.java]: @longtimeago Actually it is still needed as you can see here https://github.com/yegor256/takes/blob/5a5968552bb9b0de1ad4b94c1a07cc6450b6405c/src/main/java/org/takes/facets/fork/FkHitRefresh.java#L142

Comment (longtimeago) [inline src/main/java/org/takes/facets/fork/FkHitRefresh.java]: @essobedo can't we pass `this.last` into `private static void touch` ?

Comment (essobedo) [inline src/main/java/org/takes/facets/fork/FkHitRefresh.java]: @longtimeago as you like

**Comment by longtimeago** [comment]
@essobedo much better now :)
Thanks!

**Comment by longtimeago** [comment]
@rultor merge

**Comment by essobedo** [comment]
@longtimeago thank you for the review

**Comment by longtimeago** [comment]
@essobedo you are welcome :)

**Comment by essobedo** [comment]
@yegor256 please confirm

**Comment by essobedo** [comment]
@yegor256 please confirm

**Comment by essobedo** [comment]
@yegor256 This PR is waiting for a confirmation for more than a week now, so could you please do something about it? confirm it or reject it but please do something about it.

File: src/main/java/org/takes/facets/fork/FkHitRefresh.java [hunk 2]
```diff
@@ -180,4 +173,15 @@ private void touch() throws IOException {
         }
     }
 
+    /**
+     * Create the file to touch.
+     * @return The file to touch
+     * @throws IOException If fails
+     */
+    private static File createFileToTouch() throws IOException {
```
Comment (yegor256) [inline src/main/java/org/takes/facets/fork/FkHitRefresh.java]: @essobedo check this out: http://www.yegor256.com/images/books/elegant-objects/seven-pages.pdf

**Comment by yegor256** [comment]
@essobedo one comment above

**Comment by essobedo** [comment]
@yegor256 thank you for the review and the remark. Please have a look to the new version of the PR. Thank you in advance.

**Comment by essobedo** [comment]
@yegor256 please review the PR

**Comment by essobedo** [comment]
@yegor256 please review the PR

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by essobedo** [comment]
@yegor256 thx

**Comment by davvd** [comment]
@elenavolokhova please, help us to evaluate quality here, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor please deploy

**Comment by elenavolokhova** [comment]
@longtimeago 
According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
 Only one issue was found during review.
Please confirm that you will try to find at least three problems next time.

**Comment by longtimeago** [comment]
@elenavolokhova I confitm

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable here.

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks for the QA review, we'll work better next time

**Comment by davvd** [comment]
@longtimeago paied 10 mins to @elenavolokhova for QA review (payment ID is `79180755`); Much appreciated! **22 mins** was added to your account, payment ID `79180778`, time spent is 382 hours and 14 mins; you're getting extra minutes here (c=7); +22 to your rating, your total score is [+6596](http://www.netbout.com/b/35024?open=rating)

**Pull request merged**